chore: add quick start guide and rework some documents (#172)
openmfp/kubernetes-graphql-gateway: docs/quickstart.md

> # Quick start

Reviewer: for titles use uppercase in the first letter of major words, nouns, verbs -> Check this [short summary on Tech Writing](https://app.mural.co/t/sapcloudplatform5636/m/sapitservices2737/1697445310046/264c068efba6bd808a49dc7ab1b00b94a6b7eebb?wid=25-1697445322028)
```suggestion
# Quick Start
```

Author: Done for all entries

> [back to the main readme](../README.md)

Reviewer: ```suggestion
[Back to the main readme](../README.md)
```

You can use:
- Option A: [KCP](https://docs.kcp.io/kcp/main/setup/quickstart/)
- Option B: standard Kubernetes cluster(e.g. [kind](https://kind.sigs.k8s.io/), minikube, etc.)

## Usage
1. Clone the repo and change to the directory:
```shell
git clone git@github.com:openmfp/kubernetes-graphql-gateway.git && cd kubernetes-graphql-gateway
```
2. Setup the environment:
```shell
# this will disable authorization
LOCAL_DEVELOPMENT=true 

# kcp is enabled by default, in case you want to run it against a standard kubernetes cluster:
ENABLE_KCP=false

# you must point to the kubeconfig of the cluster you want to run against:
KUBECONFIG=YOUR_KUBECONFIG_PATH
```
3. Run the Listener:
```shell
task listener
```
This command must result in the files being created in the `./bin/definitions` directory.
Each file corresponds to a workspace in KCP or a standard Kubernetes cluster.

Gateway will watch the directory for changes and update the schema accordingly.

4. Run the Gateway:
```shell
task gateway
```
Check the console output for the URLs of the GraphQL endpoints.

For each file in the `./bin/definitions` directory, a separate GraphQL endpoint will be created.

> ## Queries examples

Reviewer: ```suggestion
## Queries Examples
```

> For both KCP and standard kubernetes clusters you can use [configMap queries](./configmap_queries.md)

Reviewer: ```suggestion
For both KCP and standard kubernetes clusters you can use [configMap queries](./configmap_queries.md).
```

Author: Applied

> If you use a standard kubernetes cluster, you can use [pod queries](./pod_queries.md)

Reviewer: ```suggestion
If you use a standard kubernetes cluster, you can use [pod queries](./pod_queries.md).
```

> ## Custom queries

Reviewer: ```suggestion
## Custom Queries
```

> Aside from queries and mutations that represent the CRUD operations, the Gateway also has [custom queries](./custom_queries.md)

Reviewer: ```suggestion
Aside from queries and mutations that represent the CRUD operations, the Gateway also has [custom queries](./custom_queries.md).
```

Author: Applied

## Subscriptions

You can subscribe to events using the following instructions:
- [Subscriptions](./subscriptions.md)

> 

Reviewer: ```suggestion
```

Author: Done

> ## Subscriptions
> 
> You can subscribe to events using the following instructions:
> - [Subscriptions](./subscriptions.md)
> 
> 

Reviewer: Bullet points are useful for lists, for single elements it actually makes it harder to red
```suggestion
## Subscriptions

You can subscribe to events using the instructions provided in the [Subscriptions Guide](https://github.com/openmfp/kubernetes-graphql-gateway/pull/172/subscriptions.md).

```

Author: Applied

## Authorization

If you run the gateway with `LOCAL_DEVELOPMENT=false`, you need to add the `Authorization` header:
```shell
{
  "Authorization": "YOUR_TOKEN"
}
```

> [back to the main readme](../README.md)

Reviewer: ```suggestion
[Back to the main readme](../README.md)
```

Author: I would put it to the top, since sometimes you are just jumping over the docs, or can misclick, and it will cover the usecase when user want to go back right away.

Author: I pushed changes where I kept only top ref, let me know if you want to move it to bottom

Author: Great, so this one is resolved then
